src/Bicep.Core/Semantics/Compilation.cs

```diff
@@ -49,27 +50,31 @@ public Compilation(IFeatureProviderFactory featureProviderFactory, INamespacePro
         public INamespaceProvider NamespaceProvider { get; }
 
         public SemanticModel GetEntrypointSemanticModel()
-            => GetSemanticModel(SourceFileGrouping.EntryPoint);
+            // entry point semantic models are guaranteed to cast successfully
+            => (SemanticModel)GetSemanticModel(SourceFileGrouping.EntryPoint);
```

Comment:
Won't the generic overload handle this for you? `=> GetSemanticModel<SemanticModel>(SourceFileGrouping.EntryPoint)`

Reply:
Yeah I'll clean it up with other fixes here.